spec/lib/administrate/search_spec.rb

```diff
@@ -0,0 +1,72 @@
+require "spec_helper"
+require "support/constant_helpers"
+require "administrate/fields/string"
+require "administrate/fields/email"
+require "administrate/fields/number"
+require "administrate/search"
+
+class MockDashboardClass
+  ATTRIBUTE_TYPES = {
+    name: Administrate::Field::String,
+    email: Administrate::Field::Email,
+    phone: Administrate::Field::Number,
+  }
+end
+
+class MockResolver
```

Comment:
Instead of defining `MockResolver`, what do you think about using a double?

```ruby
resolver = double(resource_class: User, dashboard_class: MockDashboardClass)
```

That would let us get around defining an entire `MockResolver` class, and would be more flexible if we wanted to change the behavior of the resolver on a test-by-test basis.